### PR TITLE
Upgrade to react 15.6.2.

### PR DESCRIPTION
**Pull request by gilbode**
#3656 - upgrade to react 15.6.2.  Switches from BSD + patents to MIT license.  

https://reactjs.org/blog/2017/09/25/react-v15.6.2.html

### CODE REVIEW

**Comment by shockey** [comment]
@ponelat, bump - please approve when your side of things are in step with this PR :smile:

**Comment by shockey** [comment]
Thanks @gilbode!

**Comment by gautampriya** [comment]
This leaves in place [react-addons-perf](https://mvnrepository.com/artifact/org.webjars.npm/react-addons-perf) which is under BSD+patents. Should it not also be removed?

File: package.json [hunk 1]
```diff
@@ -56,12 +56,12 @@
     "memoizee": "0.4.1",
     "promise-worker": "^1.1.1",
     "prop-types": "^15.5.10",
-    "react": "^15.4.0",
+    "react": "^15.6.2",
     "react-addons-perf": "^15.4.0",
```
Comment (gautampriya) [inline package.json]: This leaves in place [react-addons-perf](https://mvnrepository.com/artifact/org.webjars.npm/react-addons-perf) which is under BSD+patents. Should it not be removed?

Comment (shockey) [reply package.json]: @gautampriya you're correct, this was addressed in https://github.com/swagger-api/swagger-ui/pull/5229

**Pull request merged**